engine/src/main/java/org/hibernate/search/query/engine/impl/LuceneHSQuery.java

```diff
@@ -282,6 +279,25 @@ protected void extractFacetResults() {
 		return SUPPORTED_PROJECTION_CONSTANTS;
 	}
 
+	@Override
+	protected List<IndexManager> getIndexManagers(EntityIndexBinding binding) {
+		List<IndexManager> indexManagers = super.getIndexManagers( binding );
+
+		for ( IndexManager indexManager : indexManagers ) {
+			if ( !( indexManager instanceof DirectoryBasedIndexManager ) ) {
+				//TODO clarify intent:
+				// A) improve the error message so that people understand what they should do
+				// B) Is the point really to not support "directory-based" or rather non-Lucene native based ones?
```

Comment:
Hmmm, it looks weird to limit this to directory based indexes. I would expect this method to work even for RAM based indexes.

I then realized that it was already existing code but I thought I would mention it anyway.